src/resources/movies.ts

```diff
@@ -0,0 +1,275 @@
+  });
+
+  res.status(200).json(movies);
+  });
+  
+  /**
+   * @swagger
+   * /movies/genres/{name}:
+   *   get:
+   *     summary: Get a list of movies by genre
+   *     description: Retrieve a list of movies by a specific genre from the database.
+   *     parameters:
+   *       - in: path
+   *         name: name
+   *         required: true
+   *         description: The name of the genre to filter movies by.
+   *         schema:
+   *           type: string
+   *     responses:
+   *       200:
+   *         description: List of movies retrieved successfully.
+   *         content:
+   *           application/json:
+   *             schema:
+   *               type: array
+   *               items:
+   *                 $ref: '#/components/schemas/Movie'
+   *       404:
+   *         description: Movie with title was not found.
+   */
+  getMoviesByGenreRouter.get('/:name', async (req: Request, res: Response): Promise<void> => {
+    const { name } = req.params;
+
+    const movies = await Movie.find({ genre: name })
+    .then((movies) => {
+      return movies;
+    })
+    .catch((err: Error) => {
+      throw new BadRequestError({code: 404, message: err.message });
+    });
+    
+        res.status(200).json(movies);
+  });
+  
+  
+  /**
+   * @swagger
+   * /movies/{title}:
+   *   post:
+   *     summary: Add a new movie
+   *     description: Add a new movie to the database.
+   *     parameters:
+   *       - in: path
+   *         name: title
+   *         required: true
+   *         description: The title of the movie.
+   *         schema:
+   *           type: string
+   *     requestBody:
+   *       description: Movie object to add.
+   *       required: true
+   *       content:
+   *         application/json:
+   *           schema:
+   *             type: object
+   *             properties:
+   *               title:
+   *                 type: string
+   *                 example: The Matrix
+   *               genre:
+   *                 type: array
+   *                 items:
+   *                   type: string
+   *                 example: ["Action", "Adventure", "Science Fiction"]
+   *               releaseDate:
+   *                 type: string
+   *                 example: 1999-03-31
+   *               description:
+   *                 type: string
+   *                 example: A classic sci-fi movie.
+   *     responses:
+   *       201:
+   *         description: Movie added successfully.
+   *       400:
+   *         description: Title was not provided.
+   *       404:
+   *         description: Movie was not added.
+   */
+  
+  addNewMovieRouter.post('/:title', async (req: Request, res: Response): Promise<void> => {
+    const { title } = req.params;
+    const newMovie = new Movie({
+      "title": title,
+      "genre": ["Adventure", "Science Fiction"],
+      "releaseDate": new Date('2023-01-01'),
+      "description": 'This is an example movie description.',
+    });
```

Comment:
You just get title from request body but other data just mocked. Also there is no any validations about genre. Probably need to check if gerge of movie exists in separate Genre document?

Reply:
Why do this check if the data is mocked, and we can't pass the incorrect data?

Reply:
You creating newMovie here providind data not from request, but hardcoded here with same data. How o create new movie with this data for example?
![image](https://github.com/MaxSihua/Node.js_Course_Practice/assets/113895351/0e5729c3-07cc-433c-9bf2-d8ff230491ea)

Maybe it should be like:

```suggestion
    const { title, genre, releaseDate, description } = req.body;
    const newMovie = new Movie({
      "title": title,
      "genre": genre,  
      "releaseDate": releaseDate,
      "description": description,
    });
```

Reply:
done